test/integration/api/custom-attribute-span-propogation.tap.js

```diff
@@ -0,0 +1,225 @@
+'use strict'
+
+const tap = require('tap')
+
+const API = require('../../../api')
+const AttributeFilter = require('../../../lib/config/attribute-filter')
+const helper = require('../../lib/agent_helper')
+
+const DESTINATIONS = AttributeFilter.DESTINATIONS
+
+tap.test('#addAttribute', (t) => {
+  t.autoend()
+
+  let agent = null
+  let api = null
+
+  t.beforeEach((done) => {
+    agent = helper.loadMockedAgent({
+      distributed_tracing: {
+        enabled: true
+      }
+    })
+    api = new API(agent)
+    done()
+  })
+
+  t.afterEach((done) => {
+    helper.unloadAgent(agent)
+    done()
+  })
+
+  t.test('should add attribute to current span', (t) => {
+    helper.runInTransaction(agent, () => {
+      api.startSegment('segment1', true, () => {
+        api.addCustomAttribute('key1', 'value1')
+
+        const customSpanAttributes = getCustomSpanAttributes(agent)
+        t.equal(customSpanAttributes.key1, 'value1')
+
+        t.end()
+      })
+    })
+  })
+
+  t.test('should overwrite for same key added via addCustomAttribute', (t) => {
+    helper.runInTransaction(agent, () => {
+      api.startSegment('segment1', true, () => {
+        api.addCustomAttribute('key1', 'value1')
+        api.addCustomAttribute('key1', 'last-wins')
+
+        const customSpanAttributes = getCustomSpanAttributes(agent)
+        t.equal(customSpanAttributes.key1, 'last-wins')
+
+        t.end()
+      })
+    })
+  })
+
+  t.test('should not overwrite for same key added via addCustomSpanAttribute', (t) => {
```

Comment:
minor: It wasn't clear to me that this was testing adding a transaction custom attribute with same key as custom span attribute.  A more clear description might be `should not overwrite span attribute for same key added via addCustomAttribute`.

Reply:
fair. i was prob lazily trying to squish into one line.